sdk/monitor/azure-monitor-query/src/samples/java/com/azure/monitor/query/SplitQueryByRowSample.java

```diff
@@ -0,0 +1,113 @@
+     *
+     * @param originalQuery The original query string.
+     * @param queryTimeIntervals The list of time intervals.
+     * @return A {@link LogsBatchQuery} object equivalent to the original query.
+     */
+    static LogsBatchQuery createBatchQueryFromTimeRanges(String originalQuery,
+                                                         List<QueryTimeInterval> queryTimeIntervals) {
+        LogsBatchQuery batchQuery = new LogsBatchQuery();
+
+        for (QueryTimeInterval timeInterval : queryTimeIntervals) {
+            batchQuery.addWorkspaceQuery(workspaceId, originalQuery, timeInterval);
+        }
+
+        return batchQuery;
+    }
+
+    /**
+     * Helper method to create a list of time intervals for a batch query based on the row count limit.
+     *
+     * @param originalQuery The original query string.
+     * @param maxRowPerBatch The maximum row count per batch. If multiple log entries returned in the original query
+     *                       have the exact same timestamp, the row count per batch may exceed this limit.
+     * @return A list of {@link QueryTimeInterval} objects.
+     */
+    static List<QueryTimeInterval> createQueryTimeIntervalsForBatchQueryByRowCount(String originalQuery,
+                                                                                   int maxRowPerBatch) {
+
+        String findBatchEndpointsQuery = String.format(
+            "%1$s | sort by TimeGenerated desc | extend batch_num = row_cumsum(1) / %2$s | summarize batchStart=min(TimeGenerated) by batch_num | sort by batch_num desc | project batchStart",
+            originalQuery,
+            maxRowPerBatch);
+
+        LogsQueryResult result = client.queryWorkspace(workspaceId, findBatchEndpointsQuery, QueryTimeInterval.ALL);
+        List<LogsTableRow> rows = result.getTable().getRows();
+        List<OffsetDateTime> offsetDateTimes = new ArrayList<>();
+        List<QueryTimeInterval> queryTimeIntervals = new ArrayList<>();
+
+        for (LogsTableRow row : rows) {
+            row.getColumnValue("batchStart").ifPresent(rowValue -> {
+                offsetDateTimes.add(rowValue.getValueAsDateTime());
+            });
+        }
+
+
+        for (int i = 0; i < offsetDateTimes.size(); i++) {
+            OffsetDateTime startTime = offsetDateTimes.get(i);
+            OffsetDateTime endTime = i == offsetDateTimes.size() - 1 ? OffsetDateTime.now() : offsetDateTimes.get(i + 1);
+            QueryTimeInterval timeInterval = new QueryTimeInterval(startTime, endTime);
+            queryTimeIntervals.add(timeInterval);
+        }
+
+        return queryTimeIntervals;
+    }
+
+    static LogsQueryResult simulateSingleQuery(List<LogsBatchQueryResult> batchQueryResults) {
+        List<LogsTable> logsTables = new ArrayList<>();
+        for (LogsBatchQueryResult batchQueryResult : batchQueryResults) {
+            logsTables.addAll(batchQueryResult.getAllTables());
```

Comment:
Same here - the result should not contain multiple tables. Instead, it should contain a single table with all rows from every sub-request in the batch.